docs/proposed/config-split.md

```diff
@@ -0,0 +1,450 @@
+
+Open question: do we group them into a subsection?
+
+## Repartition of the configuration options
+
+### In `tf_psa_crypto_config.h`, we have:
+#### SECTION "Platform abstraction layer"
+```
+#define MBEDTLS_FS_IO
+#define MBEDTLS_HAVE_TIME
+#define MBEDTLS_HAVE_TIME_DATE
+//#define MBEDTLS_MEMORY_DEBUG
+//#define MBEDTLS_MEMORY_BACKTRACE
+#define MBEDTLS_PLATFORM_C
+//#define MBEDTLS_PLATFORM_EXIT_ALT
+//#define MBEDTLS_PLATFORM_FPRINTF_ALT
+//#define MBEDTLS_PLATFORM_GMTIME_R_ALT
+//#define MBEDTLS_PLATFORM_MEMORY
+//#define MBEDTLS_PLATFORM_MS_TIME_ALT
+//#define MBEDTLS_PLATFORM_NO_STD_FUNCTIONS
+//#define MBEDTLS_PLATFORM_NV_SEED_ALT
+//#define MBEDTLS_PLATFORM_PRINTF_ALT
+//#define MBEDTLS_PLATFORM_SETBUF_ALT
+//#define MBEDTLS_PLATFORM_SETUP_TEARDOWN_ALT
+//#define MBEDTLS_PLATFORM_SNPRINTF_ALT
+//#define MBEDTLS_PLATFORM_TIME_ALT
+//#define MBEDTLS_PLATFORM_VSNPRINTF_ALT
+//#define MBEDTLS_PLATFORM_ZEROIZE_ALT
+//#define MBEDTLS_THREADING_ALT
+//#define MBEDTLS_THREADING_PTHREAD
+//#define MBEDTLS_THREADING_C
+
+//#define MBEDTLS_MEMORY_ALIGN_MULTIPLE      4
+//#define MBEDTLS_PLATFORM_CALLOC_MACRO        calloc
+//#define MBEDTLS_PLATFORM_EXIT_MACRO            exit
+//#define MBEDTLS_PLATFORM_FREE_MACRO            free
+//#define MBEDTLS_PLATFORM_FPRINTF_MACRO      fprintf
+//#define MBEDTLS_PLATFORM_MS_TIME_TYPE_MACRO   int64_t
+//#define MBEDTLS_PRINTF_MS_TIME    PRId64
+//#define MBEDTLS_PLATFORM_NV_SEED_READ_MACRO   mbedtls_platform_std_nv_seed_read
+//#define MBEDTLS_PLATFORM_NV_SEED_WRITE_MACRO  mbedtls_platform_std_nv_seed_write
+//#define MBEDTLS_PLATFORM_PRINTF_MACRO        printf
+//#define MBEDTLS_PLATFORM_SETBUF_MACRO      setbuf
+//#define MBEDTLS_PLATFORM_STD_CALLOC        calloc
+//#define MBEDTLS_PLATFORM_STD_EXIT            exit
+//#define MBEDTLS_PLATFORM_STD_EXIT_FAILURE       1
+//#define MBEDTLS_PLATFORM_STD_EXIT_SUCCESS       0
+//#define MBEDTLS_PLATFORM_STD_FPRINTF      fprintf
+//#define MBEDTLS_PLATFORM_STD_FREE            free
+//#define MBEDTLS_PLATFORM_STD_MEM_HDR   <stdlib.h>
+//#define MBEDTLS_PLATFORM_STD_NV_SEED_FILE  "seedfile"
+//#define MBEDTLS_PLATFORM_STD_NV_SEED_READ   mbedtls_platform_std_nv_seed_read
+//#define MBEDTLS_PLATFORM_STD_NV_SEED_WRITE  mbedtls_platform_std_nv_seed_write
+//#define MBEDTLS_PLATFORM_STD_PRINTF        printf
+//#define MBEDTLS_PLATFORM_STD_SETBUF      setbuf
+//#define MBEDTLS_PLATFORM_STD_SNPRINTF    snprintf
+//#define MBEDTLS_PLATFORM_STD_TIME            time
+//#define MBEDTLS_PLATFORM_SNPRINTF_MACRO    snprintf
+//#define MBEDTLS_PLATFORM_TIME_MACRO            time
+//#define MBEDTLS_PLATFORM_TIME_TYPE_MACRO       time_t
+//#define MBEDTLS_PLATFORM_VSNPRINTF_MACRO    vsnprintf
+```
+
+#### SECTION "General and test configuration options"
+```
+//#define MBEDTLS_CHECK_RETURN_WARNING
+//#define MBEDTLS_DEPRECATED_REMOVED
+//#define MBEDTLS_DEPRECATED_WARNING
+//#define MBEDTLS_PSA_CRYPTO_CONFIG_FILE "psa/crypto_config.h"
```

Comment:
Following up on https://github.com/Mbed-TLS/mbedtls/pull/9236#discussion_r1640202307:

> > Presumably MBEDTLS_PSA_CRYPTO_CONFIG_FILE will become TF_PSA_CRYPTO_CONFIG_FILE and MBEDTLS_PSA_CRYPTO_USER_CONFIG_FILE will become TF_PSA_CRYPTO_USER_CONFIG_FILE. (…)
>
> I haven't renamed them yet. Not sure when we will do it. It seems to me it can be done even after the repo split.

Agreed, but please add a note to this document, because when we look at this document in isolation, it seems inconsistent to remove `psa/crypto_config.h` as a file but keep an option that refers to it (both in its name and its default value).

In other words, I think renaming these options is part of the changes specified by this architectural document. But in terms of planning, it's a distinct task, and it can be done after the files have moved to their respective repositories.

Reply:
I've renamed them eventually and added a note as these are the only options to be renamed. And indeed yes, when we do it that's another question.